modules/cockroachdb/cockroachdb.go

```diff
@@ -116,16 +163,44 @@ func addEnvs(req *testcontainers.GenericContainerRequest, opts options) {
 	req.Env["COCKROACH_PASSWORD"] = opts.Password
 }
 
-func connString(opts options, host string, port nat.Port) string {
-	params := url.Values{
-		"sslmode": []string{"disable"},
+func addTLS(ctx context.Context, container testcontainers.Container, opts options) error {
+	if opts.TLS == nil {
+		return nil
 	}
 
+	caBytes := pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: opts.TLS.CACert.Raw,
+	})
+	files := map[string][]byte{
+		"ca.crt":          caBytes,
+		"node.crt":        opts.TLS.NodeCert,
+		"node.key":        opts.TLS.NodeKey,
+		"client.root.crt": opts.TLS.ClientCert,
+		"client.root.key": opts.TLS.ClientKey,
+	}
+	for filename, contents := range files {
+		if err := container.CopyToContainer(ctx, contents, filepath.Join(certsDir, filename), 0o600); err != nil {
```

Comment:
I'll have a better look at that. I had tried that first, but it required local files whereas I had `[]byte`; guess I can always create temporary files.

Secondly, at least on my first attempt, I couldn't start Cockroach without the files being in place first.

It's likely I just need another go at it & go a little deeper. Thanks for the pointer.
Is that generally the approach to take rather than the one I've taken, is there something wrong with what's here now, so I can learn?

Reply:
No, I think this could be merged as is, and if you want, create a follow-up PR with some refinements.

There is no mandatory rule for that, some modules uses the Options pattern, some not.

Reply:
Moved to a lifecycle which is a much cleaner way to access `testcontainers.Container`, thanks for the tip